libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpVP9Reader.java

```diff
@@ -0,0 +1,240 @@
+    //         0 1 2 3 4 5 6 7
+    //        +-+-+-+-+-+-+-+-+
+    //        |I|P|L|F|B|E|V|Z| (REQUIRED)
+    //        +-+-+-+-+-+-+-+-+
+    //   I:   |M| PICTURE ID  | (RECOMMENDED)
+    //        +-+-+-+-+-+-+-+-+
+    //   M:   | EXTENDED PID  | (RECOMMENDED)
+    //        +-+-+-+-+-+-+-+-+
+    //   L:   | TID |U| SID |D| (Conditionally RECOMMENDED)
+    //        +-+-+-+-+-+-+-+-+
+    //        |   TL0PICIDX   | (Conditionally REQUIRED)
+    //        +-+-+-+-+-+-+-+-+
+    //   V:   | SS            |
+    //        | ..            |
+    //        +-+-+-+-+-+-+-+-+
+
+    int header = payload.readUnsignedByte();
+    if (!gotFirstPacketOfVP9Frame) {
+      if ((header & 0x08) == 0) {
+        Log.w(
+            TAG,
+            "First payload octet of the RTP packet is not the beginning of a new VP9 partition,"
+                + " Dropping current packet.");
+        return false;
+      }
+      gotFirstPacketOfVP9Frame = true;
+    } else {
+      // Check that this packet is in the sequence of the previous packet.
+      int expectedSequenceNumber = RtpPacket.getNextSequenceNumber(previousSequenceNumber);
+      if (packetSequenceNumber != expectedSequenceNumber) {
+        Log.w(
+            TAG,
+            Util.formatInvariant(
+                "Received RTP packet with unexpected sequence number. Expected: %d; received: %d."
+                    + " Dropping packet.",
+                expectedSequenceNumber, packetSequenceNumber));
+        return false;
+      }
+    }
+
+    // Check if optional I header is present.
+    if ((header & 0x80) != 0) {
+      int optionalHeader = payload.readUnsignedByte();
+      // Check M for 15 bits PictureID.
+      if ((optionalHeader & 0x80) != 0) {
+        if (payload.bytesLeft() < 1) {
+          return false;
+        }
+      }
+    }
+
+    // Flexible-mode not implemented.
+    checkArgument((header & 0x10) == 0, "VP9 flexible mode unsupported");
+
+    // Check if optional L header is present.
+    if ((header & 0x20) != 0) {
+      payload.skipBytes(1);
+      if (payload.bytesLeft() < 1) {
+        return false;
+      }
+      // Check if TL0PICIDX header present (non-flexible mode).
+      if ((header & 0x10) == 0) {
+        payload.skipBytes(1);
+      }
+    }
+
+    // Check if optional V header is present, Refer Section 4.2.1.
+    if ((header & 0x02) != 0) {
+      int scalabilityStructure = payload.readUnsignedByte();
+      int numSpatialLayers = (scalabilityStructure >> 5) & 0x7 ;
+      int scalabilityStructureLength =
+          ((scalabilityStructure & 0x10) != 0) ? numSpatialLayers + 1 : 0;
+
+      if ((scalabilityStructure & 0x10) != 0) {
+        if (payload.bytesLeft() < scalabilityStructureLength * SCALABILITY_STRUCTURE_SIZE) {
+          return false;
+        }
+        for (int index = 0; index < scalabilityStructureLength; index++) {
+          width = payload.readUnsignedShort();
+          height = payload.readUnsignedShort();
+        }
+      }
```

Comment:
I think it's also worth adding the code that checks `G` bit, or `N_G`. Can you test if this works?

```
      // Check G bit, skips all additional temporal layers.
      if ((scalabilityStructure & 0x08) != 0) {
        // Reads N_G.
        int picturesCount = payload.readUnsignedByte();
        payload.skipBytes(picturesCount);
      }
```